driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -502,33 +526,45 @@ public void onSet(Connection connection, Message.Response response, long latency
                                 connection.release();
                                 // Try another node
                                 logger.warn("Host {} is overloaded, trying next host.", connection.address);
-                                DriverException overloaded = new DriverException("Host overloaded");
-                                logError(connection.address, overloaded);
                                 if (metricsEnabled())
                                     metrics().getErrorMetrics().getOthers().inc();
-                                retry(false, null);
-                                return;
+                                if (retryPolicy instanceof ClientFailureAwareRetryPolicy)
```

Comment:
Seems like this code is duplicated in cases OVERLOADED, SERVER_ERROR and IS_BOOTSTRAPPING, couldn't we put it in a separate function or add it in `processRetryDecision()` to avoid duplication? Same for the metrics calls